packages/core/types/Fields.ts

```diff
@@ -94,6 +94,7 @@ export type ExternalField<
   mapRow?: (value: any) => Record<string, string | number>;
   getItemSummary?: (item: Props, index?: number) => string;
   showSearch?: boolean;
+  renderFooter?: (data: any[]) => ReactElement;
```

Comment:
The props are probably better as an object, so this can be treated as a React component. Since we then also need to name it, let's use `items` instead of `data`.

```suggestion
  renderFooter?: (props: { items: any[] }) => ReactElement;
```